### PR TITLE
OpenAPI documentation

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [na] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [na] All tests and CI builds passing

### Description

This PR makes use of [`swagger-inline`](https://github.com/readmeio/swagger-inline) to generate the API documentation in OpenAPI 3.0 format for all code merged in the `dev` branch as of writing.

`swagger-inline` takes a base OpenAPI file (`api-base.yaml`) and adds contents from documentation snippets inside JavaDoc-style comments inside the source files. The result is a self-contained file `api.json` that you can import to AWS API Gateway.

I tried to keep the documentation snippets as close as the relevant code as possible, however some copy-pasting has been necessary.

This PR does not address automatic documentation generation, although the PR brings us much closer to that stage. 

No changes to existing code are made in this PR.

### To test:

- Install `nodejs`, and `npm` or equivalent package manager.
- Install `swagger-inline` globally: `npm install -y swagger-inline` (this will change when we deal with automated documentation generation).
- From the root directory of your local copy of this repo: execute `swagger-inline "**/*.java" --base api-base.yaml --format .json > api.json`

### CODE REVIEW

**Comment by br648** [comment]
This is probably a Windows thing, but had to install globally to avoid this:

'swagger-inline' is not recognized as an internal or external command, operable program or batch file.

Using this command:

npm install -y **-g** swagger-inline

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 1]
```diff
@@ -48,6 +54,65 @@ public static void register (Service spark, String apiPrefix) {
...
+         *     default:
+         *       statusCode: 200
+         *       responseParameters:
+         *         method.response.header.Access-Control-Allow-Methods: "'GET,OPTIONS'"
+         *         method.response.header.Access-Control-Allow-Headers: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'"
+         *         method.response.header.Access-Control-Allow-Origin: "'*'"
+         *   passthroughBehavior: when_no_match
+         *   httpMethod: OPTIONS
+         *   type: http
+         */
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java:115]: @binh-dam-ibigroup How come this section is not at the top of the file as is the case with other classes which have this?

File: src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java [hunk 2]
```diff
@@ -48,6 +54,65 @@ public static void register (Service spark, String apiPrefix) {
         } else {
             LOG.warn("Not building API Gateway client.");
         }
+
+        /**
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java:58]: Missing comment?:

// Open API documentation for inherited CRUD methods

**Comment by binh-dam-ibigroup** [comment]
> This is probably a Windows thing, but had to install globally to avoid this:
> 
> 'swagger-inline' is not recognized as an internal or external command, operable program or batch file.
> 
> Using this command:
> 
> npm install -y **-g** swagger-inline

Thanks @br648 it should be `npm install -g`, not `-y`

**Review by landonreed — COMMENTED** [review]
@binh-dam-ibigroup, this is not really what I was imagining as far as associating the API documentation with the code. It's completely divorced from the Spark endpoints themselves. This really doesn't aid maintenance of the API docs. At this point, it seems better to just maintain everything in a separate file, so that it's not littering up the code.

Also, my thought for integrating the CORS/API key stuff for AWS API Gateway, was that this would be appended outside of the java code because it's not very relevant to the code/APIs themselves.

**Review by br648 — CHANGES_REQUESTED** [review]
Where this is implemented it does increase the file size quite a bit. I guess that can't be helped?

Once the api.json file is generated I guess it should be deleted/moved and doesn't therefore need adding to .gitignore?

Apart from that, just one minor comment.

**Comment by landonreed** [comment]
Closing for now following team discussion due to the lack of automation this approach provides. Further work will continue on #37.

**Pull request closed without merging**